model/src/main/java/org/projectnessie/api/http/HttpTreeApi.java

```diff
@@ -272,98 +272,40 @@ LogResponse getCommitLog(
       description =
-          "This operation takes the name of the tag to reassign and the hash and the name of a "
+          "This operation takes the name of the named reference to reassign and the hash and the name of a "
               + "named-reference via which the caller has access to that hash.")
   @APIResponses({
     @APIResponse(responseCode = "204", description = "Assigned successfully"),
     @APIResponse(responseCode = "400", description = "Invalid input, ref/hash name not valid"),
     @APIResponse(responseCode = "401", description = "Invalid credentials provided"),
-    @APIResponse(responseCode = "403", description = "Not allowed to view or assign tag"),
+    @APIResponse(responseCode = "403", description = "Not allowed to view or assign reference"),
     @APIResponse(responseCode = "404", description = "One or more references don't exist"),
-    @APIResponse(responseCode = "412", description = "Update conflict")
-  })
-  void assignTag(
-      @Parameter(
-              description = "Tag name to reassign",
-              examples = {@ExampleObject(ref = "ref")})
-          @PathParam("tagName")
-          String tagName,
-      @Parameter(
-              description = "Expected previous hash of tag",
-              examples = {@ExampleObject(ref = "hash")})
-          @QueryParam("expectedHash")
-          String oldHash,
-      @RequestBody(
-              description =
-                  "Reference hash to which 'tagName' shall be assigned to. This must be either a "
-                      + "'Branch' or 'Tag' via which the hash is visible to the caller.",
-              content =
-                  @Content(
-                      mediaType = MediaType.APPLICATION_JSON,
-                      examples = {@ExampleObject(ref = "refObj"), @ExampleObject(ref = "tagObj")}))
-          Reference assignTo)
-      throws NessieNotFoundException, NessieConflictException;
-
-  @Override
-  @DELETE
-  @Path("tag/{tagName}")
-  @Operation(summary = "Delete a tag")
-  @APIResponses({
-    @APIResponse(responseCode = "204", description = "Deleted successfully."),
-    @APIResponse(responseCode = "400", description = "Invalid input, ref/hash name not valid"),
-    @APIResponse(responseCode = "401", description = "Invalid credentials provided"),
-    @APIResponse(responseCode = "403", description = "Not allowed to view or delete tag"),
-    @APIResponse(responseCode = "404", description = "Ref doesn't exists"),
-    @APIResponse(responseCode = "409", description = "update conflict"),
+    @APIResponse(responseCode = "409", description = "Update conflict")
   })
-  void deleteTag(
+  void assignReference(
       @Parameter(
-              description = "Tag to delete",
+              description = "Reference type to reassign",
               examples = {@ExampleObject(ref = "ref")})
-          @PathParam("tagName")
-          String tagName,
+          @PathParam("referenceType")
+          Reference.ReferenceType referenceType,
```

Comment:
Could you add examples for  `referenceType` params to `openapi.yaml`?

Reply:
sure, added